File: app/src/main/java/com/starsep/sokoban/mvc/FakeGameModel.java
package com.starsep.sokoban.mvc;

import android.support.annotation.NonNull;

import com.starsep.sokoban.gamelogic.HighScore;
import com.starsep.sokoban.gamelogic.Level;
import com.starsep.sokoban.gamelogic.Move;
import com.starsep.sokoban.gamelogic.Position;

public class FakeGameModel implements GameModel {
	@Override
	public void onPush() {

	}

	@Override
	public void onMove() {

	}

	@Override
	public void onWin() {

	}

	@Override
	public void moveUp() {

	}

	@Override
	public void moveDown() {

	}

	@Override
	public void moveLeft() {

	}

	@Override
	public void moveRight() {

	}

	@Override
	public void repeatLevel() {

	}

	@Override
	public void undoMove() {

	}

	@Override
	public int levelNumber() {
		return 0;
	}

	@NonNull
	@Override
	public HighScore stats() {
		return new HighScore(0, 0, 0, 0, 0);
	}

	@NonNull
	@Override
	public Position player() {
		return new Position();
	}

	@NonNull
	@Override
	public Level level() {
		return Level.getDefaultLevel();
	}

	@Override
	public void nextLevel() {

	}

	@NonNull
	@Override
	public Move lastMove() {
		return Move.DOWN;
	}

	@Override
	public void onUndoMove() {

	}

	@Override
	public void onUndoPush() {

	}

	@Override
	public void onSecondElapsed() {

	}
}

File: app/src/main/java/com/starsep/sokoban/mvc/FakeGameModel.kt
package com.starsep.sokoban.mvc

import com.starsep.sokoban.gamelogic.HighScore
import com.starsep.sokoban.gamelogic.Level
import com.starsep.sokoban.gamelogic.Move
import com.starsep.sokoban.gamelogic.Position

open class FakeGameModel : GameModel {
    override fun onPush() {

    }

    override fun onMove() {

    }

    override fun onWin() {

    }

    override fun moveUp() {

    }

    override fun moveDown() {

    }

    override fun moveLeft() {

    }

    override fun moveRight() {

    }

    override fun repeatLevel() {

    }

    override fun undoMove() {

    }

    override fun stats(): HighScore {
        return HighScore(0, 0, 0, 0, 0)
    }

    override fun player(): Position {
        return Position()
    }

    override fun level(): Level {
        return Level.getDefaultLevel(0)
    }

    override fun nextLevel() {

    }

    override fun lastMove(): Move {
        return Move.DOWN
    }

    override fun onUndoMove() {

    }

    override fun onUndoPush() {

    }

    override fun onSecondElapsed() {

    }
}
